lib/backpex/live_components/form_component.ex

```diff
@@ -81,9 +83,12 @@ defmodule Backpex.FormComponent do
       |> drop_readonly_changes(fields, assigns)
       |> put_upload_change(socket, :validate)
 
+    metadata = Resource.build_changeset_metadata(socket.assigns, target)
+
     changeset =
-      item_action_types
-      |> Resource.change(change, fields, assigns, live_resource, target: target)
+      init_schema
+      |> changeset_function.(change, metadata)
+      |> Map.put(:action, :validate)
```

Comment:
Still not sure where to place the code that calls the changeset function and if we should move it to another module 🤔 I also did not include special calls like `before_changeset`. We even state that we do not support advanced fields like `HasMany` in item actions in our docs (of course, we should change this in the future).